pkg/deploy/deploy.go

```diff
@@ -402,34 +402,44 @@ func ApplyParams(componentPath string, imageParamsMap map[string]string, isUpdat
 	return nil
 }
 
-// SubscriptionExists checks if a Subscription for the operator exists in the given namespace.
-// if exsit, return object; if not exsit, return nil.
-func SubscriptionExists(cli client.Client, namespace string, name string) (*ofapiv1alpha1.Subscription, error) {
+// GetSubscription checks if a Subscription for the operator exists in the given namespace.
+// if exist, return object; otherwise, return error.
+func GetSubscription(cli client.Client, namespace string, name string) (*ofapiv1alpha1.Subscription, error) {
 	sub := &ofapiv1alpha1.Subscription{}
 	if err := cli.Get(context.TODO(), client.ObjectKey{Namespace: namespace, Name: name}, sub); err != nil {
-		if apierrs.IsNotFound(err) {
-			return nil, nil
-		}
+		// real error or 'not found' both return here
 		return nil, err
 	}
-
 	return sub, nil
 }
 
+// Delete given Subscription if it exists
+// Do not error if the Subscription does not exist.
+func DeleteExistingSubscription(cli client.Client, operatorNs string, subsName string) error {
+	sub, err := GetSubscription(cli, operatorNs, subsName)
+	// found subscription to delete
+	if sub != nil {
+		if err := cli.Delete(context.TODO(), sub); err != nil {
+			return fmt.Errorf("error deleting subscription %s: %w", sub.Name, err)
+		}
+		return nil
+	}
+	// return nil if not found or error to Get it
+	return client.IgnoreNotFound(err)
+}
```

Comment:
It might be a matter of personal preference, as the code does the same thing, though I feel this is a bit easier to read/scan:

```suggestion
func DeleteExistingSubscription(cli client.Client, operatorNs string, subsName string) error {
	sub, err := GetSubscription(cli, operatorNs, subsName)
	if err != nil {
		return client.IgnoreNotFound(err)
	}
	
	if err := cli.Delete(context.TODO(), sub); client.IgnoreNotFound(err) != nil {
		return fmt.Errorf("error deleting subscription %s: %w", sub.Name, err)
	}
	
	return nil
}
```